### ISSUE TITLE
Validation on test set

### ISSUE THREAD

**Issue by andreaslentz** [carolchenyx/MGFN.#25]
Hi, I notice in your code that you are testing during training on the test set in every step.  What is the reason behind this? 
Because normal machine learning practice is to test the model on a validation set and only calculate the loss during training.
Then the model hasn't seen the test set before you calculate the final metric - AUC. This ensures we do not overfit to the specific test set but generalize to all test sets.

Looking forward to your clarification.

From the main code:
```
if step % 1 == 0 and step > 0:
            auc, pr_auc = test(test_loader, model, args, device, savepath)
            log_writer.add_scalar('auc-roc', auc, step)
            log_writer.add_scalar('pr_auc', pr_auc, step)

            test_info["epoch"].append(step)
            test_info["test_AUC"].append(auc)
            test_info["test_PR"].append(pr_auc)
            if args.datasetname == 'XD':
                if test_info["test_PR"][-1] > best_PR:
                    best_PR = test_info["test_PR"][-1]
                    torch.save(model.state_dict(), savepath + '/' + args.model_name + '{}-i3d.pkl'.format(step))
                    save_best_record(test_info, os.path.join(savepath + "/", '{}-step-AUC.txt'.format(step)))
            else:
                if test_info["test_AUC"][-1] > best_AUC:
                    best_AUC = test_info["test_AUC"][-1]
                    torch.save(model.state_dict(), savepath + '/' + args.model_name + '{}-i3d.pkl'.format(step))
                    save_best_record(test_info, os.path.join(savepath + "/", '{}-step-AUC.txt'.format(step)))
```
